File: java-kotlin-project/app/src/main/java/example/java/dataclass/Rectangle.java
package example.java.dataclass;

public class Rectangle {
    private double width;
    private double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        if (width < 0) {
            throw new IllegalArgumentException("Wrong width!");
        }
        this.width = width;
    }

    public double getHeight() {
        return height;
    }

    public void setHeight(double height) {
        if (height < 0) {
            throw new IllegalArgumentException("Wrong height!");
        }
        this.height = height;
    }

    public double getArea() {
        return width * height;
    }
}

File: java-kotlin-project/app/src/main/java/example/kotlin/dataclass/Rectangle.kt
package example.kotlin.dataclass

class Rectangle(width: Double, height: Double) {
    var width: Double = width
        set(value) {
            require(value > 0) { "Wrong width" }
            field = value
        }

    var height: Double = height
        set(value) {
            require(value > 0) { "Wrong height" }
            field = value
        }

    val area: Double
        get() = width * height
}